packages/core/src/anchor/anchor-processing-loop.ts

```diff
@@ -30,51 +30,54 @@ export class AnchorProcessingLoop {
-      this.#anchorStoreQueue.run(streamId.toString(), async () => {
-        try {
+    this.#loop = new ProcessingLoop(logger, store.infiniteList(batchSize), async (streamId) => {
+      try {
+        logger.debug(
+          `Loading pending anchor metadata for Stream ${streamId} from AnchorRequestStore`
+        )
+        const entry = await store.load(streamId)
+        logger.debug(`Pending anchor metadata for Stream ${streamId} loaded successfully`)
+        logger.debug(
+          `Polling CAS for anchor status for commit CID ${entry.cid} of Stream ${streamId}`
+        )
+        const event = await cas.getStatusForRequest(streamId, entry.cid).catch(async (error) => {
+          logger.warn(`No request present on CAS for ${entry.cid} of ${streamId}: ${error}`)
+          const requestCAR = await eventHandler.buildRequestCar(streamId, entry.cid)
+          const creationResult = await cas.create(new AnchorRequestCarFileReader(requestCAR))
           logger.debug(
-            `Loading pending anchor metadata for Stream ${streamId} from AnchorRequestStore`
+            `Successfully created request on the CAS for commit CID ${entry.cid} of Stream ${streamId}`
           )
-          const entry = await store.load(streamId)
-          logger.debug(`Pending anchor metadata for Stream ${streamId} loaded successfully`)
-          logger.debug(
-            `Polling CAS for anchor status for commit CID ${entry.cid} of Stream ${streamId}`
-          )
-          const event = await cas.getStatusForRequest(streamId, entry.cid).catch(async (error) => {
-            logger.warn(`No request present on CAS for ${entry.cid} of ${streamId}: ${error}`)
-            const requestCAR = await eventHandler.buildRequestCar(streamId, entry.cid)
-            const creationResult = await cas.create(new AnchorRequestCarFileReader(requestCAR))
-            logger.debug(
-              `Successfully created request on the CAS for commit CID ${entry.cid} of Stream ${streamId}`
-            )
-            return creationResult
+          return creationResult
+        })
+        logger.debug(
+          `Anchor status for commit CID ${entry.cid} of Stream ${streamId} polled successfully`
+        )
+        logger.debug(
+          `Handling anchor event with status ${event.status} for commit CID ${entry.cid} of Stream ${streamId}`
+        )
+        const isTerminal = await eventHandler.handle(event)
+        logger.debug(
+          `Anchor event with status ${event.status} for commit CID ${entry.cid} of Stream ${streamId} handled successfully`
+        )
+        if (isTerminal) {
+          logger.debug(`Removing entry from AnchorRequestStore for Stream ${streamId}`)
+          // Remove iff tip stored equals to the tip we processed
+          // Sort of Compare-and-Swap.
+          await this.#anchorStoreQueue.run(streamId.toString(), async () => {
+            const loaded = await store.load(streamId)
```

Comment:
bummer about the extra state store read, but agreed this is the simplest fix without a transactional state store.  LGTM

Reply:
It is actually negligibly fast on qa-private node. Bummer, but a tolerable one.